docs/guides/ssr-and-hygration.md

```diff
@@ -0,0 +1,198 @@
+---
+title: SSR and Hydration
+nav: 20
+---
+
+## Server Side Rendering (SSR)
+
+Sever Side Rendering (SSR) is a technique that help us to render our components into
+HTML strings on the server, send them directly to the browser, and finally "hydrate" the
+static markup into a fully interactive app on the client.
+
+### React
+
+Let's say we want to render an stateless app using react. In order to do that we need
+to use `express`, `react` and `react-dom/server`, we don't need `react-dom/client`
+since it's a statless app.
+
+Let's dive into that:
+
+- `express` help us to built a web app that we can run using node
+- `react` help us to built the ui components that we use in our app
+- `react-dom/server` help us to render our components on server side
+
+```json
+// package.json
+{
+  "compilerOptions": {
+    "noImplicitAny": false,
+    "noEmitOnError": true,
+    "removeComments": false,
+    "sourceMap": true,
+    "target": "esnext"
+  },
+  "include": ["**/*"]
+}
+```
+
+> **Note:** try not to forget to remove all comments in your `package.json` file
+
+```tsx
+// app.tsx
+export const App = () => {
+  return (
+    <html>
+      <head>
+        <meta charSet="utf-8" />
+        <meta name="viewport" content="width=device-width, initial-scale=1" />
+        <title>Static Server Side Rendering App</title>
+      </head>
+      <body>
+        <div>Hello World!</div>
+      </body>
+    </html>
+  )
+}
+```
+
+```tsx
+// server.tsx
+import express from 'express'
+import React from 'react'
+import ReactDOMServer from 'react-dom/server'
+
+import { App } from './app.tsx'
+
+const port = Number.parseInt(process.env.PORT || '3000', 10)
+const app = express()
+
+app.get('/', (_, res) => {
+  const { pipe } = renderToPipeableStream(<App />, {
```

Comment:
Where did the `renderToPipeableStream` come from? I don't see it in the current scope in this docs :P